### ISSUE TITLE
Running on Macos?

### ISSUE THREAD

**Issue by dazzng** [Softcatala/whisper-ctranslate2#64]
Can you run this on Macos?

If so, where do I type this code: pip install -U whisper-ctranslate2

Can I open a terminal inside the folder where the files are located and run this code or do I have to install phyton?

**Comment by ytain** [comment]
It does run on macOS. However you have to use python3 so you have to install python3 via ```brew```.
Instead of ```pip``` use ```pip3```.
And yes, you need to run it in the terminal.

**Comment by dazzng** [comment]
thank you I will try this and come back to you again if problems come up

**Comment by SlavikCA** [comment]
I couldn't figure out: what are supported `--compute_type` on M1 Mac (ARM64) ?

**Comment by ytain** [comment]
Just use ```auto```. The same I use for x86-64 platform. It will autoselect the best value for specific platform.

**Comment by dazzng** [comment]
when I run the command on the folder where the recording is located, it says no command of whisper-ctranslate2 is found

**Comment by ytain** [comment]
You'll have to do ```brew install pkg-config```, cause that's what is missing for the building of PyAV per error message.

Are you on a M1 or M2 machine?

**Comment by dazzng** [comment]
I am on a m2 pro machine. 

<img width="566" alt="image" src="https://github.com/Softcatala/whisper-ctranslate2/assets/91325481/7a7b0ced-99f8-4ee4-b38f-48a602491bde">

**Comment by dazzng** [comment]
do you think it installed properly?

**Comment by ytain** [comment]
Yes, it is installed correctly. However you have outdated brew formulae which means you have to do ```brew update && brew upgrade```.

**Comment by dazzng** [comment]
ok thanks I did them but I still get this error when I run the command "pip3 install -U whisper-ctranslate2"

<img width="573" alt="Screenshot 2023-12-04 at 17 59 59" src="https://github.com/Softcatala/whisper-ctranslate2/assets/91325481/0e0e0098-0a56-4ace-b80a-bc43add1014f">

**Comment by dazzng** [comment]
also this:
<img width="565" alt="image" src="https://github.com/Softcatala/whisper-ctranslate2/assets/91325481/02aeeca4-b656-46d6-80f1-22e83893b730">

**Comment by ytain** [comment]
Sorry, that error is coming from trying to build ```av-10.0.0```. I have no idea how to help you with this error since I don't have access to arm based macs.

From what I see across the projects that do the whisper ai models for speech to text inference transcription like whisper.cpp/faster-whisper/whisper-ctranslate2, the code is running on Intel based macs primarily. 

Your best chance to have it running is to try the app called WhisperScript, it uses the same code as faster-whisper which also is based off of whisper-ctranslate2. The link for the app is in one of my above replies. WhisperScript will run natively on arm based macs.

**Comment by dazzng** [comment]
ok thank you. So Whisper script is more or less the same thing as what I am trying to do here?

**Comment by ytain** [comment]
Yes, it's a GUI based app. You don't need anymore to use the terminal.

**Comment by Purfview** [comment]
@dazzng Wasn't you running faster-whisper from my repo? This is practically the same thing, it won't work on your GPU.

**Comment by dazzng** [comment]
yeah I was just trying alternatives.

**Comment by mrfragger** [comment]
> yeah I was just trying alternatives.

I have an M1 and use whisper.cpp with large and medium Core ML models.  First time large is run it takes 15 1/2 hours till it starts and first time medium takes 2 to 3 hours if I remember correctly.  I didn't want to compile them with XCode and get a developer account so just downloaded precomipled models but still takes that extra "first time run".

I get around 1.7x realtime speed on large and about 3.5x realtime speed with medium.  I use -ng (no graphics) for large since I only got 8GB RAM and it just goes a tad slower but at least I can still use the laptop.

`st=$SECONDS && for f in *.opus ; do ffmpeg -hide_banner -i "$f" -f wav -ar 16000 -ac 1 - | nice ~/whisper/whisper.cpp-1.5.4/./main -m ~/whisper/whisper.cpp-1.5.4/models/ggml-$setmodel.bin - -ovtt -of "$f" -t 8 -l "$setlanguage"  $translate -ml $maxlength -sow $setprintcolors  $setng ; for f in *.vtt ; do sed -r -i .bak -e "s|\ballah\b|Allah|g" -e's|\[BLANK_AUDIO\]||g'   "$f" ; done && for i in *opus.vtt ; do mv -i -- "$i" "$(printf '%s' "$i" | sed '1s/.opus.vtt/.vtt/')" ; [ ! -d vttsubs ] && mkdir vttsubs/ ; mv *.vtt vttsubs/ ; done && rm *.bak ; done && secs=$((SECONDS-st)); printf '\nwhisper.cpp took %02dh:%02dm:%02ds\n' $(($secs/3600)) $(($secs%3600/60)) $(($secs%60))`

**Comment by ondrejhenek** [comment]
**Hi! It was a bumpy ride, but I got this working on MacOS with an M3 chip.**

I got the same error as dazzng. In a nutshell, there is a dependency issue with a newest version of one library which bubbles all the way up to a python version. See this thread and the thread it links to for more: https://github.com/PyAV-Org/PyAV/issues/1177#issuecomment-1772487688

**My solution was to downgrade python to 3.10** which is the latest compatible version following [this SO guide](https://stackoverflow.com/questions/62898911/how-to-downgrade-python-version-from-3-8-to-3-7-mac) (written for 3.7 so just replace the numbers :) )

I hope this helps someone.
